### PR TITLE
dev/core##4146 Replace _tpl_vars with get_template_vars()

### PR DESCRIPTION
**Pull request by larssandergreen**
Overview
----------------------------------------
Just trying this find & replace out to see if tests pass, will need testing.

### CODE REVIEW

File: CRM/Core/Smarty.php [hunk 1]
```diff
@@ -300,11 +300,11 @@ public function appendValue($name, $value) {
   }
 
   public function clearTemplateVars() {
-    foreach (array_keys($this->_tpl_vars) as $key) {
+    foreach (array_keys($this->get_template_vars()) as $key) {
```
Comment (eileenmcnaughton) [inline CRM/Core/Smarty.php:303]: I don't think it is a problem in this class FWIWI

Comment (larssandergreen) [reply CRM/Core/Smarty.php:303]: Does it hurt to change it though? Haven't really looked at it in detail.

Comment (eileenmcnaughton) [reply CRM/Core/Smarty.php:303]: nah I don't think it hurts

File: ext/eventcart/CRM/Event/Cart/Form/MerParticipant.php [hunk 2]
```diff
@@ -34,7 +34,7 @@ public function appendQuickForm(&$form) {
     foreach ($custom_fields_post as $key => $field) {
       CRM_Core_BAO_UFGroup::buildProfile($form, $field, CRM_Profile_Form::MODE_CREATE, $this->participant->id);
     }
-    $custom = CRM_Utils_Array::value('custom', $form->getTemplate()->_tpl_vars, []);
+    $custom = CRM_Utils_Array::value('custom', $form->getTemplate()->get_template_vars(), []);
```
Comment (colemanw) [inline ext/eventcart/CRM/Event/Cart/Form/MerParticipant.php:37]:
```suggestion
    $custom = $form->getTemplate()->get_template_vars()['custom'] ?? [];
```

**Comment by eileenmcnaughton** [comment]
@larssandergreen  - @colemanw has now merged the compatibility layer PR to `getTemplateVars()` can be used now

**Comment by eileenmcnaughton** [comment]
@larssandergreen I feel pretty confident on this change - it direct property access to `->getTemplateVars()` - I think it the PR is cleaned up it is mergable

**Comment by seamuslee001** [comment]
@larssandergreen can you please rebase this please?

**Comment by eileenmcnaughton** [comment]
Hey @larssandergreen I'm closing this out cos I think it is otherwise addressed. I think we still have a couple of those cases with the + in the tpls but other than that I think Smarty3 is humming pretty well on latest - was def good you brought that back to life

**Pull request closed without merging**